src/pages/index.js

```diff
@@ -30,72 +32,219 @@ const previewText = previewImageModal.querySelector(".modal__preview-text");
+avatarFormValidator.enableValidation();
 
-// form popup
-const newFormPopup = new PopupWithForm("#card-modal", handleFormSubmit);
-newFormPopup.setEventListeners();
+// Card Form Popup //
 
-//  profile edit popup
-const userInfo = new UserInfo(".profile__title", ".profile__text");
-const profileEditPopup = new PopupWithForm("#edit-modal", (data) => {
-  userInfo.setUserInfo(data);
-  profileEditPopup.close();
-});
+const newCardPopup = new PopupWithForm(
+  "#card-modal",
+  handleCardFormSubmit,
+  "Create"
+);
+newCardPopup.setEventListeners();
+
+//  Profile Edit Popup //
+
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__text",
+  ".profile__image"
+);
+
+const profileEditPopup = new PopupWithForm(
+  "#edit-modal",
+  handleProfileEdit,
+  "Save"
+);
 profileEditPopup.setEventListeners();
 
-// image popup
+// Image Popup //
+
 const imagePopup = new PopupWithimage("#image-modal");
 imagePopup.setEventListeners();
 
-// section class
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      const cardEl = renderCard(item);
-      section.addItem(cardEl);
-    },
-  },
-  ".cards__list"
+// Avatar Popup //
+
+const avatarPopup = new PopupWithForm(
+  "#avatar-modal",
+  handleAvatarFormSubmit,
+  "Save"
+);
+avatarPopup.setEventListeners();
+
+// Delete Card Popup //
+
+const deleteCardPopup = new PopupWithConfirmation(
+  "#confirm-modal",
+  "Deleting..."
 );
-section.renderItems();
+deleteCardPopup.setEventListeners();
+
+// functions //
 
-// functions
 function renderCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick);
-  return card.getView();
+  const cardEl = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteclick,
+    handleLikeClick
+  );
+  section.addItem(cardEl.getView());
+}
+
+function handleCardFormSubmit(data) {
+  newCardPopup.setLoading(true);
+  api
+    .addCard(data)
+    .then((data) => {
+      renderCard(data);
+    })
+    .then(() => {
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      newCardPopup.setLoading(false, "Create");
+    });
+}
+
+function handleProfileEdit(data) {
+  profileEditPopup.setLoading(true);
+  api
+    .editUserInfo(data)
+    .then((data) => {
+      userInfo.setUserInfo(data);
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditPopup.setLoading(false);
+    });
+}
+
+function handleLikeClick(cardId, isLiked, updateLikeStatus) {
+  if (isLiked) {
+    api
+      .removeLike(cardId)
+      .then((cardData) => {
+        updateLikeStatus(cardData.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .addLike(cardId)
+      .then((cardData) => {
+        updateLikeStatus(cardData.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors